File: Src/Icm.Core/CommandLine/UndefinedOptionException.vb
Imports System.Runtime.Serialization

Namespace Icm.CommandLineTools
    Public Class UndefinedOptionException
        Inherits Exception

        Private _s As String

        Public Sub New()
            MyBase.New("Undefined option")
        End Sub

        Public Sub New(ByVal opt As String)
            MyBase.New("Undefined option " & opt)
            _s = opt
        End Sub

        Public Sub New(ByVal message As String, ByVal innerException As Exception)
            MyBase.New(message, innerException)

        End Sub

        Protected Sub New(ByVal info As SerializationInfo, ByVal context As StreamingContext)
            MyBase.New(info, context)
        End Sub

    End Class

End Namespace

File: Src/Icm.Core/CommandLine/UndefinedOptionException.cs
using System;
using System.Runtime.Serialization;

namespace Icm.CommandLineTools
{
	public class UndefinedOptionException : Exception
	{


		private string _s;
		public UndefinedOptionException() : base("Undefined option")
		{
		}

		public UndefinedOptionException(string opt) : base("Undefined option " + opt)
		{
			_s = opt;
		}

		public UndefinedOptionException(string message, Exception innerException) : base(message, innerException)
		{

		}

		protected UndefinedOptionException(SerializationInfo info, StreamingContext context) : base(info, context)
		{
		}

	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
